### PR TITLE
Reduce time to deploy to device by skipping unnecessary block erasures.

### PR DESCRIPTION
**Pull request by ColinA-MSFT**
This fix should also help to improve hardware longevity by removing unnecessary flash writes.

### CODE REVIEW

File: CLR/Debugger/Debugger.cpp [hunk 1]
```diff
@@ -622,7 +622,7 @@ bool CLR_DBG_Debugger::AccessMemory( CLR_UINT32 location, UINT32 lengthInBytes,
                 {
                     case AccessMemory_Check:
                     case AccessMemory_Read:
-                        if(deviceInfo->Attribute.SupportsXIP)
+                        if (deviceInfo->Attribute.SupportsXIP)
```
Comment (smaillet-ms) [inline CLR/Debugger/Debugger.cpp]: Please don't add whitespace there like that
The original was fine as it was, and the preferred style

File: CLR/Debugger/Debugger.cpp [hunk 2]
```diff
@@ -650,25 +650,34 @@ bool CLR_DBG_Debugger::AccessMemory( CLR_UINT32 location, UINT32 lengthInBytes,
...
+                        }
                         break;
+
                      default:
                         break;
                 }
 
 
-                if(!success)
+                if (!success)
```
Comment (smaillet-ms) [inline CLR/Debugger/Debugger.cpp]: Restore this.

**Comment by smaillet-ms** [comment]
:+1:

**Comment by mortezag** [comment]
:+1:

**Pull request merged**